textdb/textdb-textql/src/main/javacc/TextQLParser.jj

```diff
@@ -0,0 +1,664 @@
+/**
+ * JavaCCcompilation options
+ */
+options {
+  	/**
+  	 * Multiple instances of the parser might be required, thus the parser
+  	 * should not be compiled as static(made up by static methods)
+  	 */
+    STATIC = false;
+    /**
+	 * Use Unicode Input since Java's implementation of String, Character and
+	 * console input uses unicode,compiling without using UNICODE_INPUT might
+	 * lead to unexpected results if you are using the wrong encoding.
+	 * Please make sure to provide a source with unicode characters, wrapping
+	 * the input in a Reader if necessary.
+     */
+    UNICODE_INPUT = true;
+    /**
+     * The TextQL language is case-insensitive
+     */
+    IGNORE_CASE = true;
+}
+
+PARSER_BEGIN(TextQLParser)
+
+package edu.uci.ics.textdb.textql.languageparser;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.function.Consumer;
+import org.apache.commons.lang3.builder.EqualsBuilder;
+
+/**
+ * Implementation of a parser for the TextQL query language.
+ * Please check TextDb-TextQL/README.md for grammar Specification
+ * 
+ * @author Flavio Bayer
+ */
+public class TextQLParser{
+  
+	/**
+	 * Statement class and subclasses(SelectStatement, CreateViewStatement)
+	 * Each Statement class has an id. Subclasses of Statements have specific
+	 * fields related to its function. These classes have no methods, they are
+	 * used only as containers to move data.
+	 * Statement --+ SelectStatement
+	 *             + CreateViewStatement
+	 */
+    static public abstract class Statement {
+		public String id;
+      	public Statement() {
+			this(null);
+      	}
+      	public Statement(String id) {
+			this.id = id;
+      	}
+      	@Override
+      	public boolean equals(Object obj) {
+			if (obj == null) { return false; }
+			if (obj.getClass() != this.getClass()) { return false; }
+			Statement kep = (Statement) obj;
+			return new EqualsBuilder()
+						.append(id, kep.id)
+						.isEquals();
+      	}
+    }
+    static public class SelectStatement extends Statement {
+        public Boolean projectAll;
```

Comment:
Add an example of a TextQL query.  Refer to the different pieces when declaring those variables.